server/src/modules/maps/maps.service.ts

```diff
@@ -317,6 +318,33 @@ export class MapsService {
         return DtoFactory(MapDto, dbResponse);
     }
 
+    async download(mapID: number, res: Response): Promise<void> {
+        const map = await this.mapRepo.get(mapID);
+
+        if (!map) throw new NotFoundException('Map not found');
+
+        const mapData = await this.getMapFileFromStore(map.name);
+
+        if (!mapData) throw new NotFoundException(`Couldn't find bsp file for ${map.name}.bsp`);
+
+        await this.mapRepo.updateMapStats(mapID, {
+            downloads: { increment: 1 }
+        });
+
+        res.set({
+            'Content-Type': 'application/octet-stream',
+            'Content-Length': mapData.length
+        });
+        res.write(mapData, 'binary');
+        res.end();
```

Comment:
Services should handle requests, that's the controller's job. So don't pass the Response to the service, have the service response a UIntArray

Reply:
Reading up on Nest docs, we don't need to do all this anyway. Once the file storage stuff is changed to return a `StreamableFile`, we just pass that back to the controller.

Note: we're going to change this whole approach anyway to just send a bucket URL to the client. But worth implementing that file store stuff properly for any files we do end up streaming through the backend.